odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,109 @@ def _validate_module_views(self, module):
                 view._check_xml()
             except Exception as e:
                 self.raise_view_error("Can't validate view:\n%s" % e, view.id)
+
+    def fix_faulty_views(self, delete=False):
+        """ Use the get_faulty_views and take appropriate actions
+            using disable_if_possible
+            This function is never called but offers the possibility to be called from a server action
+            Please don't remove
+        """
+        for view in self:
+            if not self.view_is_valid(view):
+                faulty_views = self.get_faulty_views(view,view,self.env['ir.ui.view'])
+                # Final validation
+                self._cr.execute('SAVEPOINT end_validation')
+                faulty_views.clean_properly(delete)
+                if self.view_is_valid(view):
+                    self._cr.execute('RELEASE end_validation')
+                else:
+                    self._cr.execute('ROLLBACK TO end_validation')
+                    _logger.error("issue occurs")
+
+    def get_faulty_views(self,parent_view,view, disabled_views):
+        """ Returns the faulty views
+            To determine the faulty views, we travel the tree recursively.
+            Disable all children, then enable one by one. If there is an issue.
+            The activated node or at least of its children are faulty.
+            Disable faulty node and continue recursion.
+            This function is never called from odoo code but allow to be called from a server action.
+            Please don't remove
+        """
+        if parent_view == view:
+            self._cr.execute('SAVEPOINT test_fix_views')
+            temp_disabled_views= self.deactivate_lvl1_children(view)
+            if self.view_is_valid(parent_view):
+                if self.has_active_children(view) or len(temp_disabled_views)>1 :
```

Comment:
This seems confused

* why touch `parent_view` when we know `view` and `parent_view` are the same?
* all the view's children have been deactivated in 1480 so `has_active_children` will always be false
* if the view has a single problematic child the second test won't pass, and instead of marking the sole child as problematic it's going to disable the view itself